src/libraries/System.Runtime/tests/System.Runtime.Extensions.Tests/System/Environment.SetEnvironmentVariable.cs

```diff
@@ -262,14 +262,27 @@ public void DeleteEnvironmentVariable(EnvironmentVariableTarget target)
                 // First set the value to something and then ensure that it can be deleted.
                 Environment.SetEnvironmentVariable(varName, value);
                 Environment.SetEnvironmentVariable(varName, string.Empty);
-                Assert.Null(Environment.GetEnvironmentVariable(varName));
+                Assert.Equal(string.Empty, Environment.GetEnvironmentVariable(varName));
 
                 Environment.SetEnvironmentVariable(varName, value);
-                Environment.SetEnvironmentVariable(varName, null);
-                Assert.Null(Environment.GetEnvironmentVariable(varName));
+                Environment.SetEnvironmentVariable(varName, NullString);
+                Assert.Equal(string.Empty, Environment.GetEnvironmentVariable(varName));
+            });
+        }
+
+        [Theory]
+        [MemberData(nameof(EnvironmentTests.EnvironmentVariableTargets), MemberType = typeof(EnvironmentTests))]
+        public void DeleteEnvironmentVariable(EnvironmentVariableTarget target)
+        {
+            string varName = $"Test_DeleteEnvironmentVariable ({target})";
+            const string value = "false";
 
+            ExecuteAgainstTarget(target,
+            () =>
+            {
+                // First set the value to something and then ensure that it can be deleted.
                 Environment.SetEnvironmentVariable(varName, value);
-                Environment.SetEnvironmentVariable(varName, NullString);
+                Environment.SetEnvironmentVariable(varName, null);
                 Assert.Null(Environment.GetEnvironmentVariable(varName));
```

Comment:
should we also add a test to make sure `Environment.GetEnvironmentVariables()` does not contain the variable?

Reply:
Fixed